src/screens/regimen_setup/currentProducts/addproducts/index.tsx

```diff
@@ -0,0 +1,226 @@
+import RadioButton from 'react-native-simple-radio-button-input';
+import { Ionicons } from '@expo/vector-icons';
+import { useThemeContext } from '../../../../theme';
+import { productsNames } from '../../../../libs/products';
+import { brandNames } from '../../../../libs/products';
+
+import {
+  ModalContainer,
+  ModalTitle,
+  ModalTitleProduct,
+  CardView,
+  BrandIcon,
+  Name,
+  AvailableProducts,
+  BrandView,
+  ProductCover,
+  RadioCover,
+  BrandName,
+  ProductIcon
+} from './styles';
+
+interface SearchhProductScreenProp {
+  testID?: string;
+  isVisible: boolean;
+  onBackdropPress(): void;
+}
+
+const KEYS_TO_FILTERS = ['name'];
+
+export default function SearchProductScreen(props: SearchhProductScreenProp) {
+  const { colors } = useThemeContext();
+
+  const [search, setSearch] = useState({
+    searchBrand: '',
+    searchProduct: '',
+    displayBrand: false,
+    displayProduct: false,
+    displayBrandInput: 'show',
+    displayProductInput: 'show',
+    selectedProduct: null,
+    selectedBrand: null,
+  });
+
+  const toggleSelectedProduct = (index) => {
+     setSearch({
+          ...search,
+          selectedProduct: index
+        })
+  };
+
+  const toggleSelectedBrand = (index) => {
+  setSearch({
+          ...search,
+          selectedBrand: index
+        })
+  };
+
+  const searchBrandUpdated = text => {
+    setSearch({
+      ...search,
+      searchBrand: text,
+      displayBrand: true
+    });
+  };
+
+  const brandOnclick = () => {
+    setSearch({
+      ...search,
+      displayBrandInput: 'hide',
+      displayBrand: false
+    });
+  };
+
+  const brandOnblur = () => {
+    setSearch({
+      ...search,
+      displayBrandInput: 'show'
+    });
+  };
+
+  const searchProductUpdated = text => {
+    setSearch({
+      ...search,
+      searchProduct: text,
+      displayProduct: true
+    });
+  };
+
+  const productOnclick = () => {
+    setSearch({
+      ...search,
+      displayProductInput: 'hide',
+      displayProduct: false,
+    });
+  };
+
+  const productOnblur = () => {
+    setSearch({
+      ...search,
+      displayProductInput: 'show'
+    });
+  };
+
+  const filteredProducts = productsNames.filter(
+    createFilter(search.searchProduct, KEYS_TO_FILTERS)
+  );
+
+  const filteredBrands = brandNames.filter(
+    createFilter(search.searchBrand, KEYS_TO_FILTERS)
+  );
+
+  return (
+    <Modal
+      isVisible={props.isVisible}
+      animationIn="slideInUp"
+      animationOut="slideOutDown"
+      onBackdropPress={props.onBackdropPress}
+      useNativeDriver={true}
+      style={{ display: 'flex', justifyContent: 'flex-end', margin: 0, backgroundColor: colors.FONT_DARK_COLOR }}
+    >
+      <StatusBar hidden={true} />
+      <ModalContainer>
+        <ModalTitle>
+          add <ModalTitleProduct>shampoo</ModalTitleProduct>
+        </ModalTitle>
+        {search.displayBrandInput === 'show' ? (
+          <SearchInput
```

Comment:
There is a bit of code repetition here...
You can extract this section of the code to a component for re-use.